### ISSUE TITLE
Hello, I have two questions, can you answer it?

### ISSUE THREAD

**Issue by renzemeng** [JJN123/Fall-Detection#1]
1. I would like to ask how to give a label for each picture for a set of falling pictures when preparing the training set?
2. When testing a group of pictures, how to output start frame index and stop frame index of a fall action?

decoded_imgs = self.model.predict(test_data)
RE = MSE(test_data, decoded_imgs)
predicted_classes = threshold(RE, 0.5)

Is the “predicted_classes”  the label of the model's final prediction?

**Comment by titubeta** [comment]
Which function (.py file) are you referring here?

**Comment by renzemeng** [comment]
> Which function (.py file) are you referring here?

ae_exp.py

**Comment by renzemeng** [comment]
> Which function (.py file) are you referring here?

ae_exp.py    util.py

**Comment by titubeta** [comment]
The autoencoders are trained on ONLY the non-fall activities. However, during testing the videos will have BOTH falls and non-falls.
There are no explicit labels for the video frames, they are stored in the respective directories. for e.g. all the frames inside 'fall' directory has fall label and all the frames inside 'non-fall' are considered to have non-fall labels.
The autoencoders are not sequential, so you can predict their labels without worrying about start or end index.

**Comment by renzemeng** [comment]
> 自动编码器仅针对非跌倒活动进行训练。但是，在测试过程中，视频会同时出现下降和非下降。
> 视频帧没有明确的标签，它们存储在各自的目录中。例如，'fall'目录中的所有帧都具有fall标签，并且'non-fall'内的所有帧都被认为具有非fall标签。
> 自动编码器不是连续的，因此您可以预测其标签而无需担心开始或结束索引。

But when I used dae_main_test.py to test the data under the path "Fall-Data/Thermal-Dummy/Fall"，why is the above "predict_classes" output all 0?
The following is the file generated during the test.
AEComparisions/AUC/Thermal-Dunmmy/DAE-relu_tanh-Dropout-Thermal-epochs_500-hor_flip.csv ：
,ROC AUC Score,PR AUC Score
Fall1,0.64,0.21
Fall2,0.64,0.21
mean,0.64,0.21
std,0.0,0.0

**Comment by renzemeng** [comment]
> The autoencoders are trained on ONLY the non-fall activities. However, during testing the videos will have BOTH falls and non-falls.
> There are no explicit labels for the video frames, they are stored in the respective directories. for e.g. all the frames inside 'fall' directory has fall label and all the frames inside 'non-fall' are considered to have non-fall labels.
> The autoencoders are not sequential, so you can predict their labels without worrying about start or end index.

But when I used dae_main_test.py to test the data under the path "Fall-Data/Thermal-Dummy/Fall"，why is the above "predict_classes" output all 0?
The following is the file generated during the test.
AEComparisions/AUC/Thermal-Dunmmy/DAE-relu_tanh-Dropout-Thermal-epochs_500-hor_flip.csv ：
,ROC AUC Score,PR AUC Score
Fall1,0.64,0.21
Fall2,0.64,0.21
mean,0.64,0.21
std,0.0,0.0

**Comment by JJN123** [comment]
> 1. I would like to ask how to give a label for each picture for a set of falling pictures when preparing the training set?

As Shehroz has pointed out, this is all unsupervised learning, so no labels needed when training.

> 2. When testing a group of pictures, how to output start frame index and stop frame index of a fall action?

As Shehroz has pointed out, the labels for fall frames are given in .csv files, for example:

https://github.com/JJN123/Fall-Detection/blob/master/Fall-Data/Thermal-Dummy/Labels.csv
> 
> decoded_imgs = self.model.predict(test_data)
> RE = MSE(test_data, decoded_imgs)
> predicted_classes = threshold(RE, 0.5)
> 
> Is the “predicted_classes” the label of the model's final prediction?

Yes

> But when I used dae_main_test.py to test the data under the path "Fall-Data/Thermal-Dummy/Fall"，why is the above "predict_classes" output all 0?
> The following is the file generated during the test.
> AEComparisions/AUC/Thermal-Dunmmy/DAE-relu_tanh-Dropout-Thermal-epochs_500-hor_flip.csv ：
> ,ROC AUC Score,PR AUC Score
> Fall1,0.64,0.21
> Fall2,0.64,0.21
> mean,0.64,0.21
> std,0.0,0.0

What model are you using when running dae_main_test.py?  In dae_main_test.py you se the variable "pre_load" to the path to your saved model.

**Comment by renzemeng** [comment]
> > 1. I would like to ask how to give a label for each picture for a set of falling pictures when preparing the training set?
> 
> As titubeta has pointed out, this is all unsupervised learning, so no labels needed when training.
> 
> > 1. When testing a group of pictures, how to output start frame index and stop frame index of a fall action?
> 
> As titubeta has pointed out, the labels for fall frames are given in .csv files, for example:
> 
> https://github.com/JJN123/Fall-Detection/blob/master/Fall-Data/Thermal-Dummy/Labels.csv
> 
> > decoded_imgs = self.model.predict(test_data)
> > RE = MSE(test_data, decoded_imgs)
> > predicted_classes = threshold(RE, 0.5)
> > Is the “predicted_classes” the label of the model's final prediction?
> 
> Yes
> 
> > But when I used dae_main_test.py to test the data under the path "Fall-Data/Thermal-Dummy/Fall"，why is the above "predict_classes" output all 0?
> > The following is the file generated during the test.
> > AEComparisions/AUC/Thermal-Dunmmy/DAE-relu_tanh-Dropout-Thermal-epochs_500-hor_flip.csv ：
> > ,ROC AUC Score,PR AUC Score
> > Fall1,0.64,0.21
> > Fall2,0.64,0.21
> > mean,0.64,0.21
> > std,0.0,0.0
> 
> What model are you using when running dae_main_test.py? In dae_main_test.py you need to set the variable "pre_load" to the path to your saved model.

I used this model when testing, but output all 0.
https://github.com/JJN123/Fall-Detection/blob/master/Models/Thermal/DAE-relu_tanh-Dropout-Thermal-epochs_500-hor_flip.h5

**Comment by JJN123** [comment]
> > > 1. I would like to ask how to give a label for each picture for a set of falling pictures when preparing the training set?
> > 
> > 
> > As titubeta has pointed out, this is all unsupervised learning, so no labels needed when training.
> > > 1. When testing a group of pictures, how to output start frame index and stop frame index of a fall action?
> > 
> > 
> > As titubeta has pointed out, the labels for fall frames are given in .csv files, for example:
> > https://github.com/JJN123/Fall-Detection/blob/master/Fall-Data/Thermal-Dummy/Labels.csv
> > > decoded_imgs = self.model.predict(test_data)
> > > RE = MSE(test_data, decoded_imgs)
> > > predicted_classes = threshold(RE, 0.5)
> > > Is the “predicted_classes” the label of the model's final prediction?
> > 
> > 
> > Yes
> > > But when I used dae_main_test.py to test the data under the path "Fall-Data/Thermal-Dummy/Fall"，why is the above "predict_classes" output all 0?
> > > The following is the file generated during the test.
> > > AEComparisions/AUC/Thermal-Dunmmy/DAE-relu_tanh-Dropout-Thermal-epochs_500-hor_flip.csv ：
> > > ,ROC AUC Score,PR AUC Score
> > > Fall1,0.64,0.21
> > > Fall2,0.64,0.21
> > > mean,0.64,0.21
> > > std,0.0,0.0
> > 
> > 
> > What model are you using when running dae_main_test.py? In dae_main_test.py you need to set the variable "pre_load" to the path to your saved model.
> 
> I used this model when testing, but output all 0.
> https://github.com/JJN123/Fall-Detection/blob/master/Models/Thermal/DAE-relu_tanh-Dropout-Thermal-epochs_500-hor_flip.h5

Where are you using the predicted_classes? I cannot find the code you are referring to. What data are you testing on? In other words, what did you set the value of the variable "dset" to in dae_main_test.py?
